File: javascript/bob/bob.js
module.exports = class Bob {
  constructor () {
    this.responses = {
      question: 'Sure.',
      yell: 'Whoa, chill out!',
      nothing: 'Fine. Be that way!',
      whatever: 'Whatever.'
    }
  }

  hey (utterance) {
    if (utterance.toUpperCase() === utterance && /[A-Z]/.test(utterance)) {
      return this.responses.yell
    } else if (utterance.trim().endsWith('?')) {
      return this.responses.question
    } else if (utterance.trim() === '') {
      return this.responses.nothing
    } else {
      return this.responses.whatever
    }
  }
}

File: elixir/bob/bob.exs
defmodule Bob do
  import String

  def hey(input) do
    cond do
      ends_with? input, "?" -> "Sure."
      (upcase input) == input and String.match? input, ~r/\p{Lu}/ -> "Whoa, chill out!"
      (trim input) == "" -> "Fine. Be that way!"
      true -> "Whatever."
    end
  end
end
